iib/workers/tasks/build.py

```diff
@@ -1170,3 +1178,20 @@ def _adjust_csv_annotations(operator_csvs, package_name, organization):
             csv_annotations[annotation] = value
 
         operator_csv.dump()
+
+
+def _add_ocp_label_to_index(ocp_version, temp_dir, dockerfile_name):
+    """
+    Add ocp delivery label to the dockerfile provided.
+
+    :param str ocp_version: the ocp version that the index is delivery content for.
+    :param str temp_dir: the temp directory to look for the dockerfile.
+    :param str dockerfile_name: the dockerfile name.
+    """
+    with open(os.path.join(temp_dir, dockerfile_name), 'a') as dockerfile:
+        dockerfile.write('\nLABEL com.redhat.index.delivery.version="{0}"'.format(ocp_version))
+        log.debug(
+            'Updated the dockerfile: %s with the following content:\n%s',
+            dockerfile_name,
+            dockerfile.read(),
+        )
```

Comment:
If you're going to log the contents, you need to open the file for reading as well (`a+`). After writing to the file, you need go back to the beginning of the file to read it.
```suggestion
    with open(os.path.join(temp_dir, dockerfile_name), 'a+') as dockerfile:
        dockerfile.write('\nLABEL com.redhat.index.delivery.version="{0}"'.format(ocp_version))
        dockerfile.seek(0)
        log.debug(
            'Updated the dockerfile: %s with the following content:\n%s',
            dockerfile_name,
            dockerfile.read(),
        )
```

I think a better alternative is to just log the line you are adding to the Dockerfile.